src/Lumi/Components2/Clip.purs

```diff
@@ -0,0 +1,64 @@
+module Lumi.Components2.Clip where
+
+import Prelude
+import Data.Foldable (traverse_)
+import Data.Nullable as Nullable
+import Effect (Effect)
+import Effect.Unsafe (unsafePerformEffect)
+import Lumi.Components (LumiComponent, lumiComponent, lumiElement)
+import Lumi.Components.Spacing (Space(..))
+import Lumi.Components2.Button (_linkStyle, button)
+import Lumi.Styles (styleModifier_, toCSS)
+import Lumi.Styles.Clip as Styles.Clip
+import Lumi.Styles.Theme (useTheme)
+import React.Basic.DOM as R
+import React.Basic.Emotion as E
+import React.Basic.Events (handler_)
+import React.Basic.Hooks (JSX, element, readRefMaybe, useRef, useState, (/\))
+import React.Basic.Hooks as React
+import Web.DOM (Element)
+import Web.DOM.Element as Element
+
+type ClipProps
+  = ( content :: Array JSX
+    )
+
+clip :: LumiComponent ClipProps
+clip =
+  unsafePerformEffect do
+    lumiComponent "Clip" defaults \props -> React.do
+      theme <- useTheme
+      ref <- useRef Nullable.null
+      copied /\ setCopied <- useState false
+      let
+        setCopiedOn = setCopied \_ -> true
+
+        setCopiedOff = setCopied \_ -> false
+
+        onCopy = do
+          el <- readRefMaybe ref
+          traverse_ copyElementContents (Element.fromNode =<< el)
+          setCopiedOn
```

Comment:
I feel like we should expose this behavior as a hook? Seems pretty useful to have it portable to other components.

Reply:
Good point!